### PR TITLE
NEW: type script wrapper generator

### PR DESCRIPTION
**Pull request by melsomino**

### CODE REVIEW

File: toncli/src/binding/ts.rs [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+            ts.push_str(", ");
+        }
+        ts.push_str(&format!("\n        {}: ", param.name));
+        generate_type_decl(&param.value, "", ts)?;
+        is_first = false;
+    }
+    if !is_first {
+        ts.push_str(", ");
+    }
+    ts.push_str("\n        responseHandler?: ResponseHandler\n    ): Promise<");
```
Comment (AlexeyVavilin) [inline toncli/src/binding/ts.rs:211]: Include `responseHandler` parameter for only  functions using callback for events emiting

File: toncli/src/binding/mod.rs [hunk 2]
```diff
@@ -12,40 +12,37 @@
...
 
 pub fn command(args: &[String]) -> Result<(), CliError> {
     let command_line = CommandLine::parse(args)?;
-    let mut args = command_line.args.iter();
     let api = ton_client::client::get_api_reference(Arc::new(ClientContext::new(None)?))?.api;
-    let name = args.next().map(|x| x.as_str()).unwrap_or("");
-    let item = ApiItem::from_name(&api, name)?;
     match Lang::parse(&command_line)? {
-        Lang::JS => binding_js(&api, item)?,
+        Lang::TS => binding_ts(&api, Output::parse(&command_line)?)?,
```
[the remark this answers is not in the document]
Comment (melsomino) [reply toncli/src/binding/mod.rs:45]: binding gen ported to TS and moved to tools folder

File: toncli/src/binding/ts.rs [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+ * See the License for the specific TON DEV software governing permissions and
+ * limitations under the License.
+ *
+ */
+
+use crate::errors::CliResult;
+use crate::text_generator::Output;
+use api_info::{ConstValue, Field, Function, Module, Type, API};
+
+const MODULES_HEADER: &str = r#"
```
[the remark this answers is not in the document]
Comment (melsomino) [reply toncli/src/binding/ts.rs:19]: binding gen ported to TS and moved to tools folder

File: toncli/src/binding/ts.rs (hunk 1 above)
Comment (melsomino) [reply toncli/src/binding/ts.rs:211]: binding gen ported to TS and moved to tools folder

File: toncli/src/binding/ts.rs [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+}
+
+fn generate_type(_api: &API, _module: &Module, ty: &Field, ts: &mut String) -> CliResult<()> {
+    ts.push_str(&format!("export type {} = ", ty.name));
+    generate_type_decl(&ty.value, "", ts)?;
+    ts.push_str(";\n\n");
+    Ok(())
+}
+
+fn generate_function(
```
[the remark this answers is not in the document]
Comment (melsomino) [reply toncli/src/binding/ts.rs:192]: binding gen ported to TS and moved to tools folder

**Pull request merged**